src/langchain/llm-chat.ts

```diff
@@ -0,0 +1,117 @@
+import { BaseChatModel, BaseChatModelParams } from 'langchain/chat_models/base';
+import {
+  BaseChatMessage,
+  ChatResult,
+  MessageType,
+  SystemChatMessage,
+} from 'langchain/schema';
+import { CallbackManagerForLLMRun } from 'langchain/callbacks';
+import { InvalidInputError } from '../errors.js';
+import { GenAIModel, GenAIModelOptions } from './llm.js';
+import { GenerateOutput } from '../client-types.js';
+import { concatUnique } from '../helpers/common.js';
+import { SemiRequired } from '../helpers/types.js';
+
+interface GenAILLMOutput {
+  tokenUsage: Pick<
+    GenerateOutput,
+    'input_token_count' | 'generated_token_count'
+  >;
+}
+
+export type RolesMapping = SemiRequired<
+  Record<
+    MessageType,
+    {
+      name: string;
+      stopSequence: string;
+    }
+  >,
+  'system'
+>;
+
+type Options = BaseChatModelParams &
+  GenAIModelOptions & {
+    rolesMapping: RolesMapping;
+  };
```

Comment:
The roles mapping is quite confusing. Especially the `stopSequence` which is really the `startSequence` for a message type when you check the implementation. Also, I don't where the role `name` will appear.